### PR TITLE
Added a method that gets all changes for an app

### LINKED ISSUE #55: Add ListAppChanges method
This method should list all the changes for a given app. It will be used when looking at an app to get a list of recent changes

e.g.

```protoc
  rpc ListAppChanges({app name}) returns ({list of changes});
```

### PR DESCRIPTION
**Pull request by dylanratcliffe**
Fixes #55

### CODE REVIEW

**Review by DavidS-om — APPROVED** [review]

File: changes.proto [hunk 1]
```diff
@@ -145,6 +146,17 @@ message DeleteAppRequest {
 
 message DeleteAppResponse {}
 
+// list all changes for an app
+message ListAppChangesRequest {
+  // The UUID of the app to list changes for
+  bytes UUID = 1;
```
[the remark this answers is not in the document]
Comment (dylanratcliffe) [reply changes.proto:152]: As in `AppRef` rather than UUID?

[the remark this answers is not in the document]
Comment (DavidS-om) [reply changes.proto:152]: yes. it a property is pointing at a `Thing` by UUID, call the property `ThingRef`, or `ThingID`

**Comment by DavidS-om** [comment]
please do not commit the gen directory as submodules

File: changes.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (dylanratcliffe) [reply changes.proto:152]: And we'd do that everywhere. So for example `Thing` would have a property called `ID` which just happens to be a UUID?

[the remark this answers is not in the document]
Comment (DavidS-om) [reply changes.proto:152]: Yeah, or soon we'll have no idea what all these UUID fields are pointing at

[the remark this answers is not in the document]
Comment (dylanratcliffe) [reply changes.proto:152]: Mind reviewing? https://github.com/overmindtech/sdp/issues/62

**Comment by dylanratcliffe** [comment]
Yeah that was an accident, reverting

**Pull request merged**